### ISSUE TITLE
[autocomplete] Command palette input not announced by screen readers

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4247]
# Bug report

## Current behavior

The "command palette" variant has an input at the top of the dialog. This is not announced by VoiceOver. Virtual focus lands on the first option in the list and remains there even when keyboard focus moves to the input. There is no announcement of the autocomplete behaviour or input placeholder text, only of the list of options and that you can use arrow keys to move between them.

In NVDA using Edge, I was able to move to the input but only after moving focus _backwards_ to the close button and then using arrow keys. This is not intuitive and means other screen reader users may not be informed of the available functionality.

## Expected behavior

The input is announced by screen readers. 
Keyboard and screen reader focus can move to the input. 

Ideally, focus should move to the **input** when the dialog opens. This is a combobox with autocomplete behaviour similar to the [Aria Authoring Practices Guide (APG) Editable Combobox With List Autocomplete Example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/) except that it remains in an open/expanded state. 

It should be possible to move focus between options and to/from the input itself.

## Reproducible example

Steps to replicate:
1. Turn on VoiceOver on a Mac
2. Navigate to the [Autocomplete: Command Palette example](https://base-ui.com/react/components/autocomplete#command-palette)
3. Activate the "Open command palette" button
4. Note the input (which has an accessible name) is never announced, nor is the autocomplete behaviour
5. Use the keyboard to try and move focus to/from the input (e.g. using arrow keys, tab and shift+tab)

## Base UI version

v1.2.0

## Which browser are you using?

Chrome

## Which OS are you using?

Mac OS

## Which assistive tech are you using (if applicable)?

Voiceover 

## Additional context

Removing the `aria-activedescendent` by default may fix the behaviour. 

Ensure the focus order is in a sensible order. E.g. Input > List items > Keyboard commands group > close button
